File: examples/ruby/send_bulk_sms.rb
# frozen_string_literal: true

# Send many SMS in a single request with the Pushfy Ruby SDK.
#
#   PUSHFY_API_TOKEN=... ruby send_bulk_sms.rb
#
# Env vars:
#   PUSHFY_API_TOKEN  Messaging Bearer token (required)
#
# Each message carries its own ext_id so you can reconcile delivery status
# per recipient later via pushfy.messages.status(ext_id: ...).

require "pushfy"

def main
  pushfy = Pushfy::Client.new(api_token: ENV.fetch("PUSHFY_API_TOKEN"))

  recipients = [
    { to: "5511999999999", text: "Hi Ana, your code is 4821",   ext_id: "bulk-ana" },
    { to: "5511999999999", text: "Hi Bruno, your code is 7719", ext_id: "bulk-bruno" },
    { to: "5511999999999", text: "Hi Carla, your code is 3355", ext_id: "bulk-carla" }
  ]

  result = pushfy.sms.send_bulk(recipients)

  puts "Accepted #{recipients.size} messages: #{result.inspect}"
rescue KeyError => e
  warn "Missing environment variable: #{e.message}"
  exit 1
rescue Pushfy::RateLimitError
  warn "Rate limited — back off and retry."
  exit 1
rescue Pushfy::PushfyError => e
  warn "Failed: #{e.status} #{e.code} #{e.response.inspect}"
  exit 1
end

main

File: examples/typescript/send-bulk-sms.ts
/**
 * send-bulk-sms.ts — Send many SMS in a single request.
 *
 * Run:  PUSHFY_API_TOKEN=... npx ts-node send-bulk-sms.ts
 */
import { Pushfy, SmsBulkItem, MessageSendResult } from '@pushfy/pushfy';

async function main(): Promise<void> {
  const apiToken = process.env.PUSHFY_API_TOKEN;
  if (!apiToken) throw new Error('Set PUSHFY_API_TOKEN in the environment.');

  const pushfy = new Pushfy({ apiToken });

  const messages: SmsBulkItem[] = [
    { to: '5511999999999', text: 'Hi Ana',   extId: 'bulk-ana-001' },
    { to: '5511999999999', text: 'Hi Bruno', extId: 'bulk-bruno-001' },
    { to: '5511999999999', text: 'Hi Carla', extId: 'bulk-carla-001' },
  ];

  const result: MessageSendResult = await pushfy.sms.sendBulk(messages);

  console.log(`Accepted ${result.length} message(s).`);
  for (const item of result) {
    console.log(`  ${item.ext_id ?? '(no ext_id)'} -> id=${item.id ?? '?'} phone=${item.phone ?? '?'}`);
  }
}

main().catch((err: unknown) => {
  console.error('Failed to send bulk SMS:', err instanceof Error ? err.message : err);
  process.exit(1);
});
